src/app/action-items/action-items.component.ts

```diff
@@ -56,8 +53,28 @@ export class ActionItemsComponent implements OnInit {
       .afterClosed()
       .subscribe(item => {
         if (!!item) {
+          this.cd.detectChanges();
           this.retrieveActionItems();
         }
       });
   }
+  private getActionItem(): void {
+    this.actionItem = this.tasksService.getActionItem(this.actionItemId);
+  }
+  getQueryParams(): void {
+    this.isLoadingActionItem = true;
+    this.subscription = this.route.queryParams.subscribe(params => {
+      this.actionItemId = params.id;
+      this.getActionItem();
```

Comment:
**SUGGESTION:** I think that it would be better to call it only if the `actionItemId` is defined. If it's not defined then just assign action item to `undefined` - the case with default route where nothing is selected.